### ISSUE TITLE
Transform all warnings into errors

### ISSUE THREAD

**Issue by staticdev** [adrienverge/yamllint#355]
I didn't find a way to do that. I tried -s but it is still giving warnings.

I want to run the pre-commit hook with some change in the .yamllint that does that. Is it possible?

**Comment by k0ste** [comment]
With default profile many of rules is already have error `level`. You can check that strict work via `yamllint -d relaxed` and then add `--strict`and see the difference.

Cheers

**Comment by staticdev** [comment]
@k0ste sorry but I didn't find a solution yet. `--strict` does not transform all warnings in errors as I want. I run on a sample project and still get things like:

```
warning  truthy value should be one of [false, true]  (truthy)
```

**Comment by k0ste** [comment]
@staticdev please, provide test data.

**Comment by staticdev** [comment]
test.yml

```yml
test: on
```

They keep being warnings even with `--strict`:

```
workspace/test
 % yamllint test.yml --strict
test.yml
  1:1       warning  missing document start "---"  (document-start)
  1:7       warning  truthy value should be one of [false, true]  (truthy)
```

One thing I noticed is that the status code changes from 0 to 2. I think it is just a matter of showing them as errors then.

**Comment by k0ste** [comment]
Yes, in your tests you can check `rc`, and when `!=0` - fail your tests.

**Comment by staticdev** [comment]
@k0ste yes, but this behavior is inconsistent visually. I know I can make it work on the CI, but when the developers are checking what is wrong (that should mean errors), they only see warnings. I am not sure if it is intended to be this way, seems like a bug.

**Comment by k0ste** [comment]
@staticdev, I suggest make a your `.yamllint` file, and put on the root of project. yamllint and editors will be use it automatically.

For example, `.yamllint` is:
```yaml
---
extends: default
rules:
  document-start:
    level: error
```
Result:
```python
√ yamllint-test % yamllint test.yml
test.yml
  1:1       warning  missing document start "---"  (document-start)

√ yamllint-test % mv .yamllint1 .yamllint
√ yamllint-test % yamllint test.yml
test.yml
  1:1       error    missing document start "---"  (document-start)
```

**Comment by staticdev** [comment]
@k0ste exactly, this behavior is what I expected from `--strict`.

**Comment by k0ste** [comment]
Actually, @jsok is the author of `--strict` mode. @jsok how do you think, strict mode should cover not only exit code, but change loglevel too?

**Comment by jsok** [comment]
@staticdev 

> `--strict` does not transform all warnings in errors as I want

The [documentation](https://yamllint.readthedocs.io/en/stable/configuration.html#errors-and-warnings) doesn't suggest it should/would, does it?

There's two orthogonal things here:

1. A project needs to decide what level of severity a particular rule should be (e.g. is a missing `---` a warning or an error?)
2. A project needs to decide if it wants to fail on errors only, and allow warnings to slip past (`--no-warnings` will help filter out the noise), **or** whether it will not tolerate any warnings (the behaviour of `--strict`).

If you wish to *transform* every rule of level warning into a level of error, then the only way to currently do that (to my knowledge) is via the `level` option. So you need write a custom configuration modifying any rules to be `level: error` which you think should be errors.

Neither `--no-warnings` or `--strict` though can or should IMHO "transform" all warnings into `level: error`. It sounds like what you're asking for is a new feature?

**Comment by staticdev** [comment]
@jsok long time to respond, but yes, if there is no way this is a feature request issue.
